pytest_fixtures/subscription_fixtures.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+
+from robottelo.config import settings
+from robottelo.constants import RHEL_7_MAJOR_VERSION
+from robottelo.rhsso_utils import run_command
+
+
+def unsubscribe():
+    """unregisters a machine from cdn"""
+    run_command('subscription-manager unregister')
+    run_command('subscription-manager clean')
+
+
+@pytest.fixture(scope='session')
+def clean_rhsm():
+    """removes pre-existing candlepin certs and resets RHSM."""
+    # removing the katello-ca-consumer
+    run_command('rpm -qa | grep katello-ca-consumer | xargs -r rpm -e')
```

Comment:
There is a `ContentHost.remove_katello_ca` method that can be used here, which includes a `subscription-manager clean`.

I think it would be far more reliable for the fixture to store the original rhsm.conf file before the test, than to try and mangle the file in-place.  This has the potential to destroy the rhsm.conf at a session scope, creating difficult to debug test failures.

Reply:
@pondrejk why not write an issue to track this request if you do not have time to do it in this sprint?

Reply:
I figure this didn't mark as outdated because the lines in question were removed after line 18, rhsm config editing wasn't necessary. The method Mike mentions should imho be also elevated into some fixture, I don't want to dig into fixture refactoring in this PR